lib/iris/tests/test_analysis.py

```diff
@@ -351,102 +351,170 @@ def test_multi_coord_mdtol(self):
 
 
 class TestAggregators(tests.IrisTest):
-    def test_percentile_1d(self):
+
+    def _check_coord_properties(self, cube, collapse_coord, unit, points):
+        if not isinstance(collapse_coord, list):
+            collapse_coord = [collapse_coord]
+        name = 'percentile_over_' + '_'.join(collapse_coord)
+        self.assertTrue(cube.coord(name))
+        self.assertEqual(cube.coord(name).units, unit)
+        self.assertEqual(cube.coord(name).points, points)
+        
+    def _check_collapsed_percentile(self, cube, percents, collapse_coord,
+                                    expected_result, coord_check=False,
+                                    **kwargs):
+
+        expected_result = np.array(expected_result, dtype=np.float32)
+        result = cube.collapsed(collapse_coord, iris.analysis.PERCENTILE,
+                                percent=percents, **kwargs)
+        np.testing.assert_array_almost_equal(result.data, expected_result)
+        if coord_check:
+            self._check_coord_properties(result, collapse_coord, 1, [percents])
+
+    def _check_percentile(self, data, axis, percents, expected_result,
+                          coord_check=False, **kwargs):
+
+        result = iris.analysis._percentile(data, axis, percents, **kwargs)
+
+        np.testing.assert_array_almost_equal(result, expected_result)
+
+    def test_percentile_invalid_percentile_method(self):
         cube = tests.stock.simple_1d()
+        msg = 'Unknown percentile_method'
 
-        first_quartile = cube.collapsed('foo', iris.analysis.PERCENTILE,
-                                        percent=25)
-        np.testing.assert_array_almost_equal(first_quartile.data,
-                                             np.array([2.5], dtype=np.float32))
-        self.assertCML(first_quartile, ('analysis',
```

Comment:
I see you've dropped all the CML checks from these tests. Can you re-implement them please, as they're important for checking the resultant cubes as a whole entity are as expected following a collapse operation. You will need to add new CML results for the new tests you've added as well. 

I wouldn't expect the CML to change for the existing tests (though you've renamed the tests). If the CML is changing after accounting for the renamed tests that may well be cause for concern.

Reply:
I've put these back in and added new results files for the fast_percentile_method tests.